### ISSUE TITLE
SegmentedControl SSR problem?

### ISSUE THREAD

**Issue by xorander00** [mantinedev/mantine#1280]
### What package has an issue

@mantine/core

### Describe the bug

Using Next.js v12.1.5 & Mantine v4.1.4. Everything works fine except for SegmentedControl. Figured it's something I might be doing, but can't narrow it down to anything specific in my code. Getting the following message server-side when rendering a page with SegmentedControl:  
  
```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at /home/me/project/node_modules/@mantine/core/cjs/components/SegmentedControl/SegmentedControl.js:49:21
    at div
    at /home/me/project/node_modules/@mantine/core/cjs/components/Box/Box.js:42:18
    at /home/me/project/node_modules/@mantine/core/cjs/components/Container/Container.js:53:21
    at div
    at /home/me/project/node_modules/@mantine/core/cjs/components/Box/Box.js:42:18
    at /home/me/project/node_modules/@mantine/core/cjs/components/Container/Container.js:53:21
    at div
    at /home/me/project/node_modules/@mantine/core/cjs/components/Box/Box.js:42:18
    at /home/me/project/node_modules/@mantine/core/cjs/components/Container/Container.js:53:21
    at AppMain (webpack-internal:///./src/components/shell/index.tsx:32:20)
    at NotificationsProvider (/home/me/project/node_modules/@mantine/notifications/cjs/NotificationsProvider/NotificationsProvider.js:59:5)
    at MantineProvider (/home/me/project/node_modules/@mantine/styles/cjs/theme/MantineProvider.js:58:3)
    at SessionProvider (/home/me/project/node_modules/next-auth/react/index.js:416:24)
    at Hydrate (/home/me/project/node_modules/react-query/lib/react/Hydrate.js:33:23)
    at QueryClientProvider (/home/me/project/node_modules/react-query/lib/react/QueryClientProvider.js:45:21)
    at App (webpack-internal:///./src/pages/_app.tsx:51:16)
    at StyleRegistry (/home/me/project/node_modules/styled-jsx/dist/index/index.js:671:34)
    at FlushEffectContainer (/home/me/project/node_modules/next/dist/server/render.js:423:37)
    at AppContainer (/home/me/project/node_modules/next/dist/server/render.js:438:29)
    at AppContainerWithIsomorphicFiberStructure (/home/me/project/node_modules/next/dist/server/render.js:469:57)
    at div
    at Body (/home/me/project/node_modules/next/dist/server/render.js:733:21)
```

Looks like it's this line, specifically when it tries useMantineDefaultProps:
```
const SegmentedControl = React.forwardRef((props, ref) => {
  const _a = styles.useMantineDefaultProps("SegmentedControl", defaultProps, props), {
    // ...rest of the code contained in React.forwardRef
  }
```  
  
I assumed maybe it has something to do with defaultProps on the way I'm using the theme provider, but it looks fine and no other components in @mantine/core are throwing any errors. It renders fine too on the client, no complaint about server/client state mismatch. When I replace SegmentedControl with just a set of regular Buttons, it doesn't throw any errors.
  
I'm not familiar with the mantine codebase at all, so I'll have to spend some more time taking a peek later. Any pointers would help though. Thanks! :)

### In which browser did the problem occur

Chrome, Firefox

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Not sure if a fix is needed in my code or @mantine/core#SegmentedControl.

**Comment by rtivital** [comment]
Thanks for reporting! @achmurali can you change useLayoutEffect to useIsomorphicEffect (exported from `@mantine/hooks`)?

**Comment by xorander00** [comment]
Yup, that fixed it. Thanks for the fast response. Amazing lib, keep up the awesome work man! :)

**Comment by boxmo** [comment]
@xorander00  Where do you change the useLayoutEffect to useIsomorphicEffect? I'm having the same issue.

**Comment by rtivital** [comment]
Fixed in 4.1.5
